nomad/job_endpoint_hook_connect.go

```diff
@@ -80,40 +80,56 @@ func connectGatewayDriverConfig(hostNetwork bool) map[string]interface{} {
 // the proper Consul version is used that supports the necessary Connect
 // features. This includes bootstrapping envoy with a unix socket for Consul's
 // gRPC xDS API, and support for generating local service identity tokens.
-func connectSidecarVersionConstraint() *structs.Constraint {
-	return &structs.Constraint{
-		LTarget: "${attr.consul.version}",
-		RTarget: ">= 1.8.0",
-		Operand: structs.ConstraintSemver,
+func connectSidecarVersionConstraint(cluster string) *structs.Constraint {
+	if cluster != structs.ConsulDefaultCluster && cluster != "" {
```

Comment:
I understand why we're injecting the `cluster` string into the constraint, but I don't understand the if condition. Won't we want this constraint for default clusters, too? Also, should this be a `&&` and not a `||`?

Reply:
We need to return the default constraint (down on line 91) when either the `consul.cluster = "default"` or `consul.cluster = ""` (which is treated as `default`). So this condition is for when the cluster is not explicitly `default` and not empty either.

Reply:
For clarity, the constraint for the default cluster doesn't have the cluster name in it. It's just `attr.consul.version` and not `attr.consul.default.version`.

Reply:
Oh I get it now, thanks for clarifying Tim.